pkg/containers/client.go

```diff
@@ -171,6 +193,35 @@ func (c *Client) addContainerWithCgroup(container containerdContainers.Container
 		PIDs:         pids,
 	}
 
+	sandbox, err := c.criRuntimeServiceClient.ListPodSandbox(ctx, &criapi.ListPodSandboxRequest{
+		Filter: &criapi.PodSandboxFilter{
+			Id: container.SandboxID,
+		},
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	if len(sandbox.Items) > 0 {
```

Comment:
I think in the `else` we can return an error, as we only support kubernetes pods anyway.

Reply:
Hm could this create a race condition though, so maybe not do this. I wonder if we can get an event stream from the CRI.

Reply:
Don't see anything similar to event stream. 

But what race condition you're thinking of, and how an event stream would help?